server/src/saros/server/console/InviteCommand.java

```diff
@@ -44,9 +51,58 @@ public void execute(List<String> args, PrintStream out) {
         log.warn("Invalid JID skipped: " + jid);
       }
 
-      sessionManager.invite(jids.get(true), "Invitation by server command");
       for (JID jid : jids.get(true)) {
-        sessionManager.startSharingReferencePoints(jid);
+
+        // --------------add contact if not already--------------
+        boolean contactFound = false;
+        for (XMPPContact contact :
+            contactsService.getAllContacts()) { // loop and check if contact was already added
+          if (contact.getBareJid().equals(jid)) {
+            contactFound = true;
+            break;
+          }
+        }
+
+        if (!contactFound) { // add contact
+          log.warn(
+              "Adding "
+                  + jid
+                  + " as new contact. You will have to reinvite him after he accepted being added to the contacts");
+          out.println(
+              "Adding "
+                  + jid
+                  + " as new contact. You will have to reinvite him after he accepted being added to the contacts"); // the previous warning wont be displayed in the console
+
+          final BiPredicate<String, String> questionDialogHandler = // error handler
+              (title, message) -> {
+                log.error(
+                    title
+                        + message.replace(
+                            "Do you want to add the contact anyway?",
+                            "It will be continued anyway"));
+                // It will always throw an Not subscribed error for any reason so it cant
+                // test if the contact exists.
+                // Reviewers please tell if its an bug or me being unexperienced
+                return true;
+              };
+
+          try {
+            contactsService.addContact(jid, null, questionDialogHandler);
+            log.info(
+                "The contact was added successfully. Please invite him again after he accepted");
+            out.println(
+                "The contact was added successfully. Please invite him again after he accepted");
+            return; // exit because the user cant accept the invite in time
+          } catch (OperationCanceledException e) {
+            log.error(e);
```

Comment:
Is this your intention to continue with the code on line 102 ? I do not think so.